api/src/main/java/org/pmiops/workbench/google/DirectoryServiceImpl.java

```diff
@@ -1,16 +1,71 @@
 package org.pmiops.workbench.google;
 
-import java.util.Arrays;
-import java.util.List;
+import static com.google.api.client.googleapis.util.Utils.getDefaultJsonFactory;
+import static com.google.api.client.googleapis.util.Utils.getDefaultTransport;
 
+import com.google.api.client.googleapis.auth.oauth2.GoogleCredential;
+import com.google.api.client.googleapis.json.GoogleJsonResponseException;
+import com.google.api.services.admin.directory.Directory;
 import com.google.api.services.admin.directory.DirectoryScopes;
+import com.google.api.services.admin.directory.model.User;
+import java.io.FileInputStream;
+import java.io.InputStream;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.List;
 import org.springframework.stereotype.Service;
 
 @Service
 public class DirectoryServiceImpl implements DirectoryService {
 
-  private static final List<String> SCOPES =
-    Arrays.asList(DirectoryScopes.ADMIN_DIRECTORY_USER_READONLY);
+  private static final String APPLICATION_NAME = "All of Us Researcher Workbench";
+  private static final List<String> SCOPES = Arrays.asList(
+      DirectoryScopes.ADMIN_DIRECTORY_USER_READONLY
+  );
+
+  private GoogleCredential getGoogleCredentialFromFile(String fileName) {
+    try (InputStream is = new FileInputStream(fileName)) {
+      return GoogleCredential.fromStream(is);
+    } catch (IOException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  private GoogleCredential createCredentialWithImpersonation() {
+    GoogleCredential credential =
+        getGoogleCredentialFromFile("./directory-service-sa.json");
```

Comment:
You may want to add a TODO mentioning that in future (once the domain for the users matches the domain for the project) we can use our application default credentials and don't need this separate key file anymore.